### DISCUSSION TITLE
Output Cohesive Zone Model Properties

### DISCUSSION THREAD

**Discussion by chunhuizhao478** [idaholab/moose#26392] [Q&A Modules: Solid mechanics]
Hi all, I have a question regarding the output of cohesive zone model (tangent jump, tangent traction etc). The rule for MOOSE to store the information along the interface is within the elements which are connected to the interface, on the side of smaller subdomain IDs. Since I need to extract the information and do post-processing, I wonder what is the best way to extract the values from exodus file. 

For example, in one of my simulation results using TRIA3, the tangent jump is stored in the TRIA elements:

![Screenshot 2023-12-21 at 1 12 44 PM](https://github.com/idaholab/moose/assets/93891863/5b7f5a25-6200-41d4-80b4-2b59baa26ebf)

How should I extract the values for this interface, I didn't find any group information provided by MOOSE or in paraview, what I currently do is to use netCDF4 to extract element values (but I would need to get the element ID myself) which is tricky sometime.
Any suggestion would be appreciated, thanks!

**Comment by jiangwen84** [comment]
If you just select the interface boundary, (uncheck the blocks), you should be able to visualize the values at interface only. I guess it should work.

**Reply by chunhuizhao478** [reply]
Hi @jiangwen84, I don't think it works, if I close the blocks, select the czm boundary only, the tangent jump (shown as elemental variable in paraview) will not be accessible anymore. So I wonder since MOOSE can determine where to store the information, is that possible that I extract this information somehow.

**Reply by jiangwen84** [reply]
If you can a way to do it using Paraview's filters, that would be easiest. 

Otherwise, maybe you have to build a lower dimensional surface and store the values on that surface. The Mortar implementation may be useful to look at.  Someone else might give you other solutions.
